site/sfguides/src/kepler_gl_maps_inside_snowflake/kepler_gl_maps_inside_snowflake.md

```diff
@@ -0,0 +1,225 @@
+
+Execute the following SQL commands in Snowflake (make sure you have the `ACCOUNTADMIN` role for these operations):
+
+```sql
+-- Run as ACCOUNTADMIN
+GRANT IMPORTED PRIVILEGES ON DATABASE OVERTURE_MAPS__TRANSPORTATION TO application DEKART__WEBGL_MAPS_FOR_SNOWFLAKE;
+
+GRANT IMPORTED PRIVILEGES ON DATABASE OVERTURE_MAPS__DIVISIONS TO application DEKART__WEBGL_MAPS_FOR_SNOWFLAKE;
+
+GRANT IMPORTED PRIVILEGES ON DATABASE OVERTURE_MAPS__PLACES TO application DEKART__WEBGL_MAPS_FOR_SNOWFLAKE;
+```
+
+These commands authorize Dekart to access the specific tables within the Overture Maps datasets, allowing it to query and display data on your map.
+
+
+<!-- ------------------------ -->
+## Create Maps with SQL in Snowflake
+Duration: 2
+
+Now we’re ready to start creating maps! Dekart allows you to visualize data directly from SQL queries, which means you can write custom queries to shape the data as you like.
+
+1. **Open the Dekart App** within Snowsight by going to **Apps** and selecting **Dekart**.
+2. **Authorize the Dekart App** with your Snowflake account.
+3. In the Dekart interface, click **Create Report** to start building your map.
+
+![Create Report in Dekart](assets/kepler_gl_snowflake_10.png)
+
+<!-- ------------------------ -->
+## Build Map in Dekart
+Duration: 5
+
+Let’s construct a map that visualizes UK highways with color-coded density of nearby EV charging stations. We’ll use SQL queries to filter and aggregate data from Overture Maps, focusing on major roads and charging infrastructure.
+
+![Build Map in Dekart](assets/kepler_gl_snowflake_11.png)
+
+### Define the UK Boundary
+
+To create a map specific to the UK, we need to define the boundaries of the UK region. This can be done using the **DIVISION_AREA** table from Overture Maps, which contains geographic boundaries for different regions.
+
+![Define UK Boundary](assets/kepler_gl_snowflake_12.png)
+
+```sql
+-- Define the UK boundary as a geographic region
+SELECT ST_ASWKT(GEOMETRY) AS GEOMETRY_WKT
+FROM OVERTURE_MAPS__DIVISIONS.CARTO.DIVISION_AREA
+WHERE COUNTRY = 'GB' AND SUBTYPE = 'country';
+```
+
+Explanation:
+- `ST_ASWKT(GEOMETRY)`: Converts the geometric shape of the UK boundary into a format compatible with Kepler.gl.
+- This query filters by `COUNTRY = 'GB'` to select only boundaries relevant to the United Kingdom and limits results to the outer boundary using `SUBTYPE = 'country'`.
+
+### Select Major Road Segments
+
+To focus on major roads, such as highways and trunk roads, we’ll query the **SEGMENT** table from Overture Maps. This will limit the map to primary roadways that are most relevant to our EV charging station analysis.
+
+![Select Major Road Segments](assets/kepler_gl_snowflake_13.png)
+
+```sql
+SELECT ST_ASWKT(s.GEOMETRY) AS GEOMETRY_WKT, s.NAMES, s.ID
```

Comment:
Should it be something like this?
```
with uk_boundary as (SELECT GEOMETRY
FROM OVERTURE_MAPS__DIVISIONS.CARTO.DIVISION_AREA
WHERE COUNTRY = 'GB' AND SUBTYPE = 'country')
SELECT ST_ASWKT(s.GEOMETRY) AS GEOMETRY_WKT, s.NAMES, s.ID
FROM OVERTURE_MAPS__TRANSPORTATION.CARTO.SEGMENT s, uk_boundary ub
WHERE ST_INTERSECTS(ub.GEOMETRY, s.GEOMETRY) AND s.CLASS IN ('motorway', 'trunk');
```

Reply:
Perhaps it was not clear that it's part of the query, not complete one. I will update.